src/redux/productsSlice.js

```diff
@@ -0,0 +1,113 @@
+import { createSlice, createAsyncThunk } from "@reduxjs/toolkit"
+import firebaseClient from "firebase/client"
+
+const initialState = {
+  data: [],
+  products: [],
+  loadingStatus: "idle",
+  error: null,
+}
+
+const productsSlice = createSlice({
+  name: "products",
+  initialState,
+  reducers: {},
+})
+
+export const createNewProduct = createAsyncThunk(
+  "products/createNewProduct",
+  async (payload) => {
+    try {
+      await _createNewProduct(
+        payload.productName,
+        payload.productPrice,
+        payload.productPhoto,
+        payload.productRoom,
+        payload.productDuration
+      )
+      return payload
+    } catch (error) {
+      console.error("error", error)
```

Comment:
These errors should `rejectWithValue` so that the state can know that there has been an error.
https://redux-toolkit.js.org/api/createAsyncThunk#handling-thunk-errors

Reply:
fixed